04-spatial-operations.Rmd

```diff
@@ -209,21 +324,60 @@ plot(joined["population_millions"], add = TRUE)
 par(mfrow = c(1, 1))
 ```
 
-<!-- I would expand it a little bit on spatial joins. Typically, this operation is also known as spatial overlay. Of course, you can also use other geometries (lines), and especially you can not only use st_intersects (another reason why the topological stuff should have been already covered before). Also interesting would be to show what happens if there are multiple matches, e.g., multiple points per polygon. In sp, this was solved using a list containing the IDs of the matched objects while not returning the corresponding attribute table. Overall, a comparison with sp::over would be also interesting -->
+This operation is also know as spatial overlay.
+`st_join()` performs by default a left join (see chapter \@ref(vector-attribute-joining), but you can change this to an inner join operation.
+Another default of `st_join()` is to intersect the two provided layers.
+Of course, you can use any other topological operations we have already encountered above such as `st_within()` or or `st_touches()` (please refer also to the help page of `st_join()`.
+In the example above, we have added features of a point layer to a polygon layer.
+Please note that there might be multiple point matches for one polygon. 
+Had we chosen to select the four (instead of three) most populous cities in the world, two of them would have belonged to China (Shanghai and Beijing, give it a try yourself).
+In such a case `st_join()` simply adds a new row.
+In our example we would end up with two polygons representing China.
+
+
+#### Non-overlapping joins 
+
+<!-- Nearest neighbour analysis -->
+<!-- e.g. two point's datasets (non-overlapping) -->
+<!-- e.g. two point's datasets (overlapping) -->
+<!-- ? topological problems of joining lines/polygons? -->
+<!-- joining different types (e.g. points + polygons = geometry) -> save as GPKG? -->
+<!-- `merge()`; `st_interpolate_aw()` -->
+
+
+<!--### Modifying geometry data; still need to change the corresponding cross-references-->
```

Comment:
I have changed the heading but I have not changed the corresponding cross-references. I wanted to wait until we agree on a final structure and header naming.